tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1050,9 +1056,11 @@ void EnqueueProgramCommand::assemble_device_commands(
                 }
             }
         }
-        // Unlink the last subcmd of the current core range
-        if (!write_linear) {
-            kernel_bins_dispatch_subcmds.back().back().flags |= CQ_DISPATCH_CMD_PACKED_WRITE_LARGE_FLAG_UNLINK;
+    }
+    // Unlink the last subcmd of every dispatch, to ensure we don't hold the path reservation for an incredible long time.
```

Comment:
Not unlinking here would have hung all subsequent path reservations issued by any other mcast in the dispatch. Should we mention this in the comment?

Reply:
Done.